jaeger-core/src/main/java/com/uber/jaeger/baggage/RemoteBaggageRestrictionManager.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2017, Uber Technologies, Inc
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.jaeger.baggage;
+
+import static com.uber.jaeger.baggage.Constants.DEFAULT_MAX_VALUE_LENGTH;
+
+import com.uber.jaeger.baggage.http.BaggageRestriction;
+import com.uber.jaeger.exceptions.BaggageRestrictionErrorException;
+import com.uber.jaeger.metrics.Metrics;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class RemoteBaggageRestrictionManager implements BaggageRestrictionManager {
+  private static final int DEFAULT_REFRESH_INTERVAL_MS = 60000;
+  private static final int DEFAULT_REFRESH_DELAY_MS = 0;
+
+  private final String serviceName;
+  private final BaggageRestrictionProxy proxy;
+  private final Timer pollTimer;
+  private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+  private final Metrics metrics;
+  private final boolean denyBaggageOnInitializationFailure;
+  private boolean initialized;
+  private Map<String, Integer> restrictions = new HashMap<String, Integer>();
+
+  public RemoteBaggageRestrictionManager(
+      String serviceName, BaggageRestrictionProxy proxy, Metrics metrics, boolean denyBaggageOnInitializationFailure) {
+    this(serviceName, proxy, metrics, denyBaggageOnInitializationFailure,
+        DEFAULT_REFRESH_INTERVAL_MS, DEFAULT_REFRESH_DELAY_MS);
+  }
+
+  public RemoteBaggageRestrictionManager(
+      String serviceName,
+      BaggageRestrictionProxy proxy,
+      Metrics metrics,
+      boolean denyBaggageOnInitializationFailure,
+      int refreshIntervalMs,
+      int refreshDelayMs
+  ) {
+    this.serviceName = serviceName;
+    this.proxy = proxy;
+    this.metrics = metrics;
+    this.denyBaggageOnInitializationFailure = denyBaggageOnInitializationFailure;
```

Comment:
This name is misleading because of the initialization race condition.

Specifically, there is a period of time after the creation of the object but before the first successful response from jaeger agent where the failure behavior applies. 

Perhaps `denyBaggageDuringInitializationOrFailure` is more descriptive. 

Also, I don't really like how we handle initialization. I'd like for object initialization not be blocking (as it is), but also have a means of signalling that the object is ready for accepting requests.  A lightweight interface with `startUp` and `shutdown` methods might make sense.